presto-native-execution/presto_cpp/main/tests/PrestoExchangeSourceTest.cpp

```diff
@@ -349,8 +349,7 @@ TEST_F(PrestoExchangeSourceTest, basic) {
     queue->noMoreSources();
 
     auto exchangeSource =
-        std::make_shared<PrestoExchangeSource>(producerUri, 3, queue);
-    exchangeSource->setMemoryPool(pool);
+        std::make_shared<PrestoExchangeSource>(producerUri, 3, queue, pool_);
```

Comment:
> // Test both with memory pool and without memory pool conditions

Do we still need this? Can we simplify the test to test only the case when pool is set?

Reply:
Right since now MemoryPool is required. I'll simplify the test